fix: adding thread status in the thread search api #34

MadaraUchiha-314 opened the pull request:
- Adds the status as an accepted field in the POST `/threads/search` API
- Refactors thread status used in multiple places to a single reference
- Fixes: #34

nfcampos commented:
The status of the thread is meant to be a field reflecting the set of runs that exists for that thread, therefore it shouldn't be updated directly. eg. creating a run would set the thread status to busy, when that run completes the thread status would update to idle, and so on

MadaraUchiha-314 commented:
@beryder Addressed your review comment

@nfcampos Understood your point and made changes to not have status in ThreadPatch.

Kindly review and merge.

MadaraUchiha-314 commented:
@beryder @nfcampos Any help to get this reviewed and merged ?

nfcampos reviewed (approved):
Thanks
